### PR TITLE
[kube-prometheus-stack] fixes thanos ruler object store configs

### PR DESCRIPTION
**Pull request by thameezb**
#### What this PR does / why we need it
Fixes ThanosRuler objectstore configs

#### Which issue this PR fixes
fixes https://github.com/prometheus-community/helm-charts/issues/3912

#### Checklist
- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/kube-prometheus-stack/templates/thanos-ruler/ruler.yaml [hunk 1]
```diff
@@ -101,7 +101,8 @@ spec:
 {{- end }}
 {{- if .Values.thanosRuler.thanosRulerSpec.objectStorageConfig }}
   objectStorageConfig:
-{{ toYaml .Values.thanosRuler.thanosRulerSpec.objectStorageConfig | indent 4 }}
+    key: objectstore-configs.yaml
```
Comment (GMartinez-Sisti) [inline charts/kube-prometheus-stack/templates/thanos-ruler/ruler.yaml:104]:
```suggestion
    key: object-storage-configs.yaml
```

File: charts/kube-prometheus-stack/templates/thanos-ruler/secret.yaml [hunk 2]
```diff
@@ -11,4 +11,7 @@ data:
 {{- if .Values.thanosRuler.thanosRulerSpec.alertmanagersConfig }}
   alertmanager-configs.yaml: {{ toYaml .Values.thanosRuler.thanosRulerSpec.alertmanagersConfig | b64enc | quote }}
 {{- end }}
+{{- if .Values.thanosRuler.thanosRulerSpec.objectStorageConfig }}
+  objectstore-configs.yaml: {{ toYaml .Values.thanosRuler.thanosRulerSpec.objectStorageConfig | b64enc | quote }}
```
Comment (GMartinez-Sisti) [inline charts/kube-prometheus-stack/templates/thanos-ruler/secret.yaml:15]:
```suggestion
  object-storage-configs.yaml: {{ toYaml .Values.thanosRuler.thanosRulerSpec.objectStorageConfig | b64enc | quote }}
```

**Review by GMartinez-Sisti — APPROVED** [review]

**Comment by GMartinez-Sisti** [comment]
Thanks @thameezb !!

**Comment by GMartinez-Sisti** [comment]
Oh no! Thanks for letting us know @dragoangel. Let me see if we can easily fix it forward as you suggested, or we’ll revert it.

**Comment by dragoangel** [comment]
> Oh no! Thanks for letting us know @dragoangel. Let me see if we can easily fix it forward as you suggested, or we’ll revert it.

Thank you for your quick reply

**Pull request merged**